db/migrate/20240319062553_create_sla_events.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSlaEvents < ActiveRecord::Migration[7.0]
+  def change
+    create_table :sla_events do |t|
+      t.references :applied_sla, null: false
```

Comment:
We could also add an `account_id`, 'inbox_id' and 'sla_id' as well to make it easier to query all the events related to one account as well as doing the filtering with out having to do a join

Reply:
we can auto populate this values into the model in before validation using `ensure_*` method pattern which we are using

Reply:
Makes sense. Done.